soroban-env-host/src/budget.rs

```diff
@@ -557,6 +580,98 @@ impl Default for BudgetImpl {
                     cpu.const_term = 3000906;
                     cpu.lin_term = ScaledU64(0);
                 }
+                ContractCostType::Bls12381EncodeFp => {
+                    cpu.const_term = 661;
+                    cpu.lin_term = ScaledU64(0);
+                }
+                ContractCostType::Bls12381DecodeFp => {
+                    cpu.const_term = 985;
+                    cpu.lin_term = ScaledU64(0);
+                }
+                ContractCostType::Bls12381G1Validate => {
+                    cpu.const_term = 732301;
+                    cpu.lin_term = ScaledU64(0);
+                }
+                ContractCostType::Bls12381G2Validate => {
+                    cpu.const_term = 1063432;
+                    cpu.lin_term = ScaledU64(0);
+                }
+                ContractCostType::Bls12381G1ProjectiveToAffine => {
+                    cpu.const_term = 92642;
+                    cpu.lin_term = ScaledU64(0);
+                }
+                ContractCostType::Bls12381G2ProjectiveToAffine => {
+                    cpu.const_term = 100742;
+                    cpu.lin_term = ScaledU64(0);
+                }
+                ContractCostType::Bls12381G1Add => {
+                    cpu.const_term = 7689;
+                    cpu.lin_term = ScaledU64(0);
+                }
+                ContractCostType::Bls12381G1Mul => {
+                    cpu.const_term = 2458985;
+                    cpu.lin_term = ScaledU64(0);
+                }
+                ContractCostType::Bls12381G1Msm => {
+                    cpu.const_term = 2426722;
+                    cpu.lin_term = ScaledU64(96397671);
+                }
+                ContractCostType::Bls12381MapFpToG1 => {
+                    cpu.const_term = 1541554;
+                    cpu.lin_term = ScaledU64(0);
+                }
+                ContractCostType::Bls12381HashToG1 => {
+                    cpu.const_term = 3211191;
+                    cpu.lin_term = ScaledU64(6713);
+                }
+                ContractCostType::Bls12381G2Add => {
+                    cpu.const_term = 25207;
+                    cpu.lin_term = ScaledU64(0);
+                }
+                ContractCostType::Bls12381G2Mul => {
+                    cpu.const_term = 7873219;
+                    cpu.lin_term = ScaledU64(0);
+                }
+                ContractCostType::Bls12381G2Msm => {
+                    cpu.const_term = 8035968;
+                    cpu.lin_term = ScaledU64(309667335);
```

Comment:
At 310M, does this Bls12381G2Msm host fn won't run at our current limit of 100M txMaxInstructions but only in future when we increase the limits?

Reply:
this is `ScaledU64`, which has extra 7 bits of precision, so the actual parameter is `309667335 >>7` which is around 2.4M.